### ISSUE TITLE
Self-hosted sync server?

### ISSUE THREAD

**Issue by cakee-ru** [TeamNewPipe/NewPipe#8039]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

since NP is detached from cloud accounts you need to manually export data if you use multiple devices. what if we add sync targets like in Joplin? since export feature already works, we just need to manage exported files. we could use nextcloud, dropbox or simple http server in docker. how relevant this feature would be? any potential hazards or caveats?

### Why do you want this feature?

I use NP both on phone and tablet. I'd really like to self-host a sync server to sync subs, playlists, watch progress and settings. everything, basically. 

### Additional information

I ask before doing any development and potentially could create both PR to main app and dockerized http server.

**Comment by hmlendea** [comment]
It'd be great to have the subscriptions and watch later lists in sync across devices (phone, tablet, TV,...)

**Comment by Mrnofish** [comment]
Generic file-based sync sounds like a good idea to me. This would leave the choice of the sync transport up to the user, enabling e.g. Syncthing that doesn't even require a dedicated server.

**Comment by cakee-ru** [comment]
alright, I'll inspect NP sources and will draw some ideas on how to implement it. thanks for commenting.

**Comment by opusforlife2** [comment]
Please have a look at this and other comments: https://github.com/TeamNewPipe/NewPipe/pull/2668#issuecomment-623369694

**Comment by Mrnofish** [comment]
> An implementation using owncloud/nextcloud can be considered though.

Both require subscriptions, and I don't think sync should be *exclusively* tied to a service that requires periodic payments.

> Storing user data, encrypted or not should be avoided as much as possible

The app already stores subscriptions, playlists, view statuses etc on each device independently, I can't imagine why it would be a problem for that data to be optionally shared between those same devices.

**Comment by BeatLink** [comment]
Has anyone looked into https://github.com/39aldo39/DecSync? It is filesystem based, designed to be able to work with any existing sync system, is completely open source and requires no accounts or signups. It is merely a file and folder based standard optimized for conflict free syncing

**Comment by cakee-ru** [comment]
> Both require subscriptions, and I don't think sync should be exclusively tied to a service that requires periodic payments.

not really, as self-hosted versions are completely free. anyway, there should be dedicated server for NP.

**Comment by Mrnofish** [comment]
> not really, as self-hosted versions are completely free

Of course the software itself is free, but the infrastructure is not.

Syncthing runs on the same devices that are kept in sync, so it doesn't require anything else beside those.

Apps like Joplin, for instance, support basically any sync-a-folder solution, leaving the choice to the user. Whether you are self-hosting, renting a server, paying for a sync service, or none of the above, you're covered.

That's great IMO, nobody is left out and everybody's happy.

**Comment by lazee486** [comment]
Found this rather than starting a new request, nextcloud has developed into a very open free self hostable project on your own vps or infrastructure, even just a snap. Either a nextcloud app like gpodder sync or something that just syncs to a nextcloud folder would be amazing.  I use nextcloud as a google replacement backend, and newpipe with syncing would eliminate YouTube from my device...

**Comment by lazee486** [comment]
Thanks, will try piped as well :)

**Comment by branexis** [comment]
How about sync using cloudflare worker in user's own account? It should be well within free limits.
